packages/gatsby/src/commands/develop-process.ts

```diff
@@ -715,6 +718,26 @@ module.exports = async (program: IDevelopArgs): Promise<void> => {
       webpackActivity = null
     }
 
+    if (isSuccessful) {
+      const state = store.getState()
+      const mapOfPagesToStaticQueryHashes = mapPagesToStaticQueryHashes(
+        state,
+        stats
+      )
+
+      const publicDir = path.join(program.directory, `public`)
+
+      mapOfPagesToStaticQueryHashes.forEach(
```

Comment:
This is naive now in the sense that it writes page data files for all pages on every change. This needs to diff and check for what changed (cheaply) and only rewrite those

Reply:
Fixed